src/crypto/keccak.c

```diff
@@ -121,11 +122,11 @@ void keccak(const uint8_t *in, size_t inlen, uint8_t *md, int mdlen)
     temp[rsiz - 1] |= 0x80;
 
     for (i = 0; i < rsizw; i++)
-        st[i] ^= ((uint64_t *) temp)[i];
+        st[i] ^= swap64le(((uint64_t *) temp)[i]);
 
     keccakf(st, KECCAK_ROUNDS);
 
-    memcpy(md, st, mdlen);
+    memcpy_swap64le(md, st, mdlen);
```

Comment:
This should check that mdlen is a multiple of sizeof(uint64_t) and call memcpy_swap64le with mdlen/sizeof(uint64_t).